Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py

```diff
@@ -1422,6 +1422,136 @@ def condition(self, fragment, stepList, key, hasHarvest):
     offset = 0.21,
 )
 
+class UpgradeWorkflow_ProdLikeRunningPU(UpgradeWorkflow_ProdLike):
+    def __init__(self, suffix, offset, fixedPU,
+         steps = [
+        'GenSimHLBeamSpot14',
+        'Digi',
+        'DigiTrigger',
+        'Reco',
+        'RecoGlobal',
+        'RecoNano',
+        'HARVEST',
+        'HARVESTGlobal',
+        'HARVESTNano',
+        'MiniAOD',
+        'ALCA',
+        'ALCAPhase2',
+        'Nano',
+        ],
+        PU = [
+            'GenSimHLBeamSpot14',
+            'Digi',
+            'DigiTrigger',
+            'Reco',
+            'RecoGlobal',
+            'RecoNano',
+            'HARVEST',
+            'HARVESTGlobal',
+            'HARVESTNano',
+            'MiniAOD',
+            'ALCA',
+            'ALCAPhase2',
+            'Nano',
+        ]):
+        super(UpgradeWorkflow_ProdLikeRunningPU, self).__init__(steps, PU, suffix, offset)
+        self.__fixedPU = fixedPU
+    def setupPU_(self, step, stepName, stepDict, k, properties):
+        #  change PU skipping ALCA and HARVEST
+        if not ('ALCA' in step or 'HARVEST' in step) and '--pileup' in stepDict[stepName][k]:
```

Comment:
Do we need to check ALCA and HARVEST here? I think it is removed already in ProdLike.

Reply:
If I remove the `ALCA`  and `HARVEST` checks I get:

```bash
  File "/home/adrianodif/running_pu/dev/bin/slc7_amd64_gcc11/runTheMatrix.py", line 622, in <module>
    ret = runSelected(opt)
  File "/home/adrianodif/running_pu/dev/bin/slc7_amd64_gcc11/runTheMatrix.py", line 31, in runSelected
    if len(undefSet)>0: raise ValueError('Undefined workflows: '+', '.join(map(str,list(undefSet))))
ValueError: Undefined workflows: 21034.21101
```

and this comes from the fact that `setup_` from `ProdLike` removes the step by setting `stepDict[stepName][k]=None`. So the steps are actually still there and I need to check if they are not `None`. I'm changing it to a more general:

```python
if stepDict[stepName][k] is not None and '--pileup' in stepDict[stepName][k]:
            stepDict[stepName][k]['--pileup'] = 'AVE_' + str(self.__fixedPU) + '_BX_25ns'
```